[Feat] 신고하기 관련 기능구현

AndBlack99 opened the pull request:
## 📌 PR 제목
[Feat] 신고하기 관련 기능구현

---

## 📄 개요
- buyer / 신고하기 기능
- admin / 신고된 목록 조회
- admin / 신고 상세조회
- admin / 신고 상태처리

---

## ✅ 작업 내용
<!-- 체크박스를 채우며 어떤 작업을 했는지 명확히 표현해주세요 -->
- [x] 기능 구현
- [ ] UI/UX 수정
- [ ] API 연동
- [ ] 버그 수정
- [x] 테스트 코드 작성
- [x] 문서 작성

---

## 📃 작업 세부 내용

<img width="1347" alt="스크린샷 2025-07-06 오후 9 18 37" src="https://github.com/user-attachments/assets/31f1f59b-db90-4763-a440-af1a20191c76" />

<img width="1351" alt="스크린샷 2025-07-06 오후 9 18 04" src="https://github.com/user-attachments/assets/95fefe37-f701-4409-880c-65ac995b249e" />

---

## 🔍 관련 이슈
#62  

---

## 📝 참고 사항
 

---

## 🙏 리뷰 요청 사항
-

Commit:
feat : 신고하기 관련 기능구현

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/report/62

Commit:
feat : 신고하기 모달 Thymeleaf 템플릿 구현

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/report/62

Commit:
refactor : 신고하기 관련 기능 사용자/관리자 분리

- 사용자/관리자 기능별로 Controller와 Service 분리
- support 패키지: 구매자용 신고 기능 (생성, 내 목록 조회)
- admins 패키지: 관리자용 신고 관리 (목록, 상세, 상태변경, 통계)

Commit:
refactor : 신고하기 관련 기능 사용자/관리자 분리

- 사용자/관리자 기능별로 Controller와 Service 분리
- support 패키지: 구매자용 신고 기능 (생성, 내 목록 조회)
- admins 패키지: 관리자용 신고 관리 (목록, 상세, 상태변경, 통계)

Commit:
refactor : 시간관리 ZonedDateTime 통일

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/report/62

Commit:
refactor : 관련없는 기능 로직 제거

Commit:
feat: 리포트 테스트 코드 및 일부 파일 수정/추가/삭제 복구

Commit:
fix : address의 Builder 초기값 설정 누락 해결

- 실수로 제거한 AddressRequestDto의 Builder.Default 어노테이션 추가

cake0420 commented on src/main/java/com/team5/catdogeats/admins/controller/AdminReportController.java:
```diff
@@ -0,0 +1,178 @@
+            log.info("관리자 신고 목록 조회: adminId={}, page={}, size={}", adminId, page, size);
+
+            // 검색 조건 DTO 생성
+            ReportSearchDto searchDto = ReportSearchDto.builder()
```
admin을 조회하는데 왜 user을 조회하나요?

AndBlack99 replied:
수정된 내용이었는데 커밋이 안됐었나봅니다.
관리자 인증 로직 수정과 함께 수정하겠습니다!

cake0420 commented:
관리자 인증 로직을 전부 수정하셔야 합니다

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/report/62

Commit:
fix: AdminReportController 세션 기반 인증으로 변경

- UserPrincipal → HttpSession + AdminControllerUtils 방식 적용
- 다른 관리자 컨트롤러와 일관성 유지

Commit:
Merge branch 'develop' of github.com:prgrms-aibe-devcourse/AIBE1-FinalProject-TodaysCommit_BE into feat/report/62

AndBlack99 committed:
Merge branch 'develop' into feat/report/62
